### PR TITLE
Update build for Saxon 10 compatibility

### PR DESCRIPTION
**Pull request by infotexture**
## Description

The docs builds have been failing since DITA-OT was upgraded to Saxon 10.5 with https://github.com/dita-ot/dita-ot/pull/3485.

## Motivation and Context

This PR upgrades the bundled Gradle plug-ins to the latest versions to allow a different Saxon version to be specified:

- [dita-ot-gradle](https://github.com/eerohele/dita-ot-gradle) 0.7.0 → 0.7.1
- [saxon-gradle](https://github.com/eerohele/saxon-gradle)  0.7.0 → 0.9.0-beta4

The Saxon Gradle Plugin requires the latest version of Gradle, so this PR upgrades the wrapper to Gradle 7.2.

## How Has This Been Tested?

- Ran docs build locally, docs output is generated as expected:

    ```shell
    ./gradlew -PditaHome=../dita-ot/src/main --no-daemon --rerun-tasks
    ```

## Type of Changes

- Bug fix _(non-breaking change which fixes an issue)_
- New feature _(non-breaking change which adds functionality)_

## Documentation and Compatibility

- What documentation changes are needed for this feature?
  _None – user command for building docs repo from source remains the same: `./gradlew …`_
- Will this change affect backwards compatibility or other users' overrides?
  _No_

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+    mavenCentral()
+}
+
+configurations {
+    saxon
+}
+
+dependencies {
+    // Use Saxon-HE 10.5 like DITA-OT, instead of the version that comes with `saxon-gradle`
+    saxon 'net.sf.saxon:Saxon-HE:10.5'
```
Comment (infotexture) [inline build.gradle:18]: @eerohele Do the changes to the build file look right to you, or is there a better way to do this?

**Comment by infotexture** [comment]
Tested DITA-OT distribution build with a [fresh Git worktree](https://github.com/dita-ot/dita-ot/wiki/Fresh-dist-worktree), updating the `docsrc/` submodule in the worktree folder to use the code from the tip of this PR branch.

Although the docs build `dist` task works fine when run alone from the docs repo, it fails when run from the parent `dist` build in the DITA-OT repo:

```shell
> Task :html FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':html'.
> The following error occurred while executing this line:
  /…/dita-ot.dist.worktree/build/tmp/dist/plugins/org.dita.base/build.xml:29: 
     The following error occurred while executing this line:
  /…/dita-ot.dist.worktree/build/tmp/dist/plugins/org.dita.base/build_init.xml:73: 
     java.lang.NoClassDefFoundError: net/sf/saxon/lib/CollationURIResolver
```

@jelovirt To get this to work, we may need to also upgrade the Gradle wrapper in the core repo, 
— but that may require additional changes there, as the current build suggests compatibility issues with Gradle 7:

```
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings
```

File: build.gradle (hunk 1 above)
Comment (eerohele) [reply build.gradle:18]: LGTM. 👍🏻

**Review by eerohele — APPROVED** [review]

**Review by jelovirt — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (jelovirt) [reply build.gradle:18]: We can merge this, but it's likely that we'll update to 10.6 before release.

**Review by infotexture — COMMENTED** [review]
Tested again with JDK 8 and the Gradle 7.2 changes in the core repo from https://github.com/dita-ot/dita-ot/pull/3803, build succeeds. ✅ 

_(Earlier errors were apparently due to incompatibilities with JDK 11)_

File: build.gradle (hunk 1 above)
Comment (infotexture) [reply build.gradle:18]: Merging as is, created separate PR https://github.com/dita-ot/docs/pull/356 for the 10.6 change.

**Pull request merged**